### PR TITLE
New MapReduce utility functions

### PR DESCRIPTION
**Pull request by cdahlqvist**

### CODE REVIEW

File: src/riak_kv_mapreduce.erl [hunk 1]
```diff
@@ -343,3 +497,19 @@ reduce_count_inputs_test() ->
...
                              none),
                         none)).
+
+reduce_limit_test() ->
+    [1,2,3] = reduce_limit([1,2,3], none),
+    [1,2] = reduce_limit([1,2,3,4,5], 2),
+    [1,2] = reduce_limit([1,2,3,4], <<"2">>),
+    [1,2] = reduce_limit([1,2,3,4], <<"{\"limit\":2}">>),
+    ?DEFAULT_REDUCE_LIMIT_SIZE = length(reduce_limit(lists:seq(1,200), none)).
+
```
Comment (evanmcc) [inline src/riak_kv_mapreduce.erl]: extra whitespace here.

File: src/riak_kv_mapreduce.erl [hunk 2]
```diff
@@ -289,6 +431,18 @@ map_object_value_test() ->
     [] = map_object_value({error, notfound}, test, <<"filter_notfound">>),
     [{error,notfound}] = map_object_value({error, notfound}, test, <<"include_notfound">>).
 
+map_datasize_test() ->
```
Comment (evanmcc) [inline src/riak_kv_mapreduce.erl:568]: could you wrap all the tests in an ifdef?  I realize this wasn't in place before, but it should have been.

**Comment by slfritchie** [comment]
Hi, Christian.  If you could find a reviewer for this old PR, that'd be helpful. The core/kv team is under the crunch right now for other review work, but it'd be wonderful you could find someone outside that team (clients?) now or find someone in the core/kv team a few weeks from now?

**Comment by slfritchie** [comment]
Pulling PRs out of the freezer ... review?

File: src/riak_kv_mapreduce.erl [hunk 3]
```diff
@@ -117,14 +141,178 @@ map_object_value_list(Acc) ->
...
+    [];
+map_delete(RiakObject, _, _) ->
+    case is_deleted(RiakObject) of
+        true ->
+            [];
+        false ->
+            {ok, C} = riak:local_client(),
+            Bucket = riak_object:bucket(RiakObject),
+            Key = riak_object:key(RiakObject),
+            C:delete(Bucket, Key),
```
Comment (seancribbs) [inline src/riak_kv_mapreduce.erl:178]: This should use the `delete_vclock` for safety.

File: src/riak_kv_mapreduce.erl [hunk 4]
```diff
@@ -117,14 +141,178 @@ map_object_value_list(Acc) ->
...
+    [];
+map_datasize(RiakObject, _, _) ->
+    [lists:foldl(fun({MD,Val}, A) ->
+                     case dict:is_key(<<"X-Riak-Deleted">>, MD) of
+                         true ->
+                             A;
+                         false ->
+                             (byte_size(Val) + A)
+                     end
+                 end, 0, riak_object:get_contents(RiakObject))].
```
Comment (seancribbs) [inline src/riak_kv_mapreduce.erl:201]: Couldn't this just use `riak_object:approximate_size/2`?

**Comment by seancribbs** [comment]
There's a decent amount of complexity in some of these functions. I'd love to see a riak_test exercising them before I sign off.

**Comment by jtuple** [comment]
Punting to Riak 2.1 development cycle

**Comment by yorkshire-steve** [comment]
Can one of the admins verify this patch?

**Pull request closed without merging**